src/main/java/de/dennisguse/opentracks/MarkerEditActivity.java

```diff
@@ -246,26 +160,20 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
                 Toast.makeText(this, R.string.marker_add_photo_canceled, Toast.LENGTH_LONG).show();
                 return;
             } else if (resultCode == RESULT_OK) {
-                setMarkerImageView(photoUri);
+                viewModel.onNewCameraPhoto(cameraPhotoUri,
```

Comment:
I cannot to figure out how. Because the URI comes from different places depending on the photo's source: camera or gallery.

Reply:
We can just assign the URI to a variable and the later use it.
```
Uri a = null;
if (resultCode == RESULT_OK) {
  a = xxx;
} else {
  a = zzz;
}
                viewModel.onNewGalleryPhoto(a,
                        viewBinding.markerEditName.getText().toString(),
                        viewBinding.markerEditMarkerType.getText().toString(),
                        viewBinding.markerEditDescription.getText().toString());
```

Reply:
Yes, but ViewModel have two methods `onNewGalleryPhoto` and `onNewCameraPhoto` because **camera URI's photo** is ready but **gallery URI's photo** need to be copied to internal OT storage.

I don't know if I'm missing something...